src/stream/src/executor/hash_agg.rs

```diff
@@ -279,23 +280,29 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
                                 input_pk_data_types.clone(),
                                 epoch,
                                 Some(hash_code),
-                                state_tables,
+                                &*state_tables.read().await,
```

Comment:
With a `RwLock`, it seems the fetch process cannot be concurrent anymore?

Reply:
We may move the apply part outside of the future, or introduce a new rwlock-wrapped state table.

Reply:
+1